crates/derive/src/combine_api.rs

```diff
@@ -128,7 +131,11 @@ impl cgo::Service for API {
                 Ok(())
             }
             _ => Err(Error::InvalidState),
-        }
+        };
+
+        let diff = mem_stats.current() - initial;
+        tracing::trace!(mem_initial = ?initial, mem_changes = ?diff, "mem changes");
```

Comment:
Just curious about our current thinking: are we planning to leave stat collection like this in as tracing, or are we going to move this to being something we pass back to Go for inclusion in Flow/Gazette backed metrics/logging?

Just as a basis for developing intuition: what values have you been seeing here for these "mem_changes" logs?

Reply:
This is a really interesting open question. Something we could do is return metrics as information messages back over the CGO bridge, as part of the extract / combine / derive protocols. This would allow us to tie memory allocations directly to catalog tasks and even shards which drove them.

Of course, these are globals and could be affected by other concurrent and unrelated operations. I have no idea whether they'd still be accurate enough to be useful under load 🤷 . I hope so.